src/parser/warlock/affliction/modules/talents/SoulConduit.js

```diff
@@ -39,11 +41,24 @@ class SoulConduit extends Analyzer {
     const avgDamage = this._totalUAdamage / (this._totalTicks > 0 ? this._totalTicks : 1);
     const shardsGained = this.soulShardTracker.getGeneratedBySpell(SPELLS.SOUL_CONDUIT_SHARD_GEN.id);
     const estimatedUAdamage = shardsGained * TICKS_PER_UA * avgDamage;
+    const totalSpent = this.soulShardTracker.spent;
+    // since we want to get the amount of shards we would MOST LIKELY get, we intentionally need to try higher values for k than usually possible
+    const { partial: probabilities } = binomialCDF(Math.round(totalSpent / 2), totalSpent, SC_PROC_CHANCE);
+    // with 15% chance per shard, it's very unlikely that we would get half of the shards refunded, so the maximum must be somewhere lower than that, and "partial" array of the return object contains those values
+    let maxP = 0;
+    let max = -1;
+    probabilities.forEach((p, k) => {
+      if (p > maxP) {
+        maxP = p;
+        max = k;
+      }
+    });
```

Comment:
this could be "simplified" to a `reduce`:

```js
// argmax(probabilities) -- if only Math.argmax existed
const max = probabilities.reduce((max, p, k, probs) => (p > probs[max]) ? k : max, 0);
```

now, whether this is desirable or not is up for debate. i'll leave which version to use up to you

Reply:
While it's an elegant one-liner, I think for the sake of readability, I'd prefer my solution (I had to look up the third and fourth argument to `reduce`)